tools/validators/instance_validator/tests/entity_instance_test.py

```diff
@@ -84,5 +84,122 @@ def testValidateBadEntityType(self):
     if instance.IsValidEntityInstance():
       self.fail('exception not raised')
 
+  def testValidateCompliantTranslation(self):
+    parsed = dict(instance_parser.parse_yaml(
+        os.path.join(_TESTCASE_PATH,
+                     'GOOD',
+                     'good_translation_compliant.yaml')))
+    entity_name = list(parsed.keys())[0]
+
+    entity = dict(parsed[entity_name])
+    instance = entity_instance.EntityInstance(entity, self.universe)
+
+    if not instance.IsValidEntityInstance():
+      self.fail('exception incorrectly raised')
+
+  def testValidateMultipleCompliantTranslation(self):
+    parsed = dict(instance_parser.parse_yaml(
+        os.path.join(_TESTCASE_PATH,
+                     'GOOD',
+                     'good_translation_multiple_compliant.yaml')))
+    entity_name = list(parsed.keys())[0]
+
+    entity = dict(parsed[entity_name])
+    instance = entity_instance.EntityInstance(entity, self.universe)
+
+    if not instance.IsValidEntityInstance():
+      self.fail('exception incorrectly raised')
+
+  def testValidateTranslationUnitValues(self):
+    parsed = dict(instance_parser.parse_yaml(
+        os.path.join(_TESTCASE_PATH,
+                     'GOOD',
+                     'good_translation_unit_values.yaml')))
+    entity_name = list(parsed.keys())[0]
+
+    entity = dict(parsed[entity_name])
+    instance = entity_instance.EntityInstance(entity, self.universe)
```

Comment:
Can the instance contents be checked? (here and in other tests too)

In general, you should at-least check that the element of the instance that's being added in the test is in the instance.  If there are possible interactions between elements, those should also be checked as well, even if they were already validated in other tests.

Reply:
Yes, the `.IsValidEntityInstance()` has all the checking/validation logic.

Reply:
Just because an instance is valid doesn't mean that it has all the contents that the config file contained.

Reply:
True, when I'm checking for `IsValidEntityInstance` though, it's after the YAML schema parse. If the YAML schema parse fails, it means there were parts of the config file that failed to adhere to the provided schema. If the YAML schema parse goes well, it means every part of the config file is accounted for (and fits into the schema). So this is why I think the instance has all the contents the config file contained.